exec/java-exec/src/main/java/org/apache/drill/exec/compile/AbstractClassCompiler.java

```diff
@@ -74,9 +76,10 @@ protected String prefixLineNumbers(String code) {
     return out.toString();
   }
 
-  protected abstract byte[][] getByteCode(ClassNames className, String sourcecode)
+  protected abstract byte[][] getByteCode(final ClassNames className, final String sourcecode)
       throws CompileException, IOException, ClassNotFoundException, ClassTransformationException;
-
+  public abstract Map<String,byte[]> compile(final ClassNames className, final String sourceCode)
```

Comment:
We spoke in person. There was some confusion about the two caches involved. One is the original class cache, which has not changed. This is a way for the compiler to give us a (name, code) map so we can match up byte codes to compiled code. This map is used only after the global class cache has decided we have a cache miss.